cookbooks/aws-parallelcluster-scheduler-plugin/resources/create_user.rb

```diff
@@ -29,6 +29,7 @@
     new_resource.system_users.each_with_index do |user, index|
       name = user[:Name]
       enable_imds = user[:EnableImds]
+      sudoer_configuration = user[:SudoerConfiguration]
```

Comment:
instead of having a separate python script, why not using a chef [template resource](https://docs.chef.io/resources/template/) to generate sudoers file for each user?

Here we are already in the loop for each user, so I think it would be enough to have something like

```
template '/etc/sudoers.d/99-parallelcluster-scheduler-plugin-#{user}' do
  source 'scheduler_plugin_user/99-parallelcluster-scheduler-plugin-sudoers-template.erb'
  variables <- pass here the sudoer_configuration (Commands and RunAs)
  [...]
end
```

where we inject the sudoer_configuration into template variable, and the template (scheduler_plugin_user/99-parallelcluster-scheduler-plugin-sudoers-template.erb) is written so that it knows what to do with the variables.

Reply:
Good idea! Changed